From 17a2fced10df27729ffbc3632e6789edf0a2609b Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 8 Dec 2017 15:54:19 +0200
Subject: [PATCH 3054/5242] MLK-17805: drm: imx: dcss: fix resume without HDMI
 cable

commit  bccae2c4b0afec116850d53c9910470dcd38f477 from
https://source.codeaurora.org/external/imx/linux-imx.git

When no HDMI cable is in, the device is runtime suspended. Hence,
there's nothing to resume in this case.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-common.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/imx/dcss/dcss-common.c b/drivers/gpu/imx/dcss/dcss-common.c
index 28b43a9..cd4afdc 100644
--- a/drivers/gpu/imx/dcss/dcss-common.c
+++ b/drivers/gpu/imx/dcss/dcss-common.c
@@ -486,6 +486,9 @@ static int dcss_resume(struct device *dev)
 	struct platform_device *pdev = to_platform_device(dev);
 	struct dcss_soc *dcss = platform_get_drvdata(pdev);
 
+	if (pm_runtime_suspended(dev))
+		return 0;
+
 	dcss_bus_freq(dcss, true);
 
 	pm_qos_add_request(&dcss->pm_qos_req, PM_QOS_CPU_DMA_LATENCY, 0);
-- 
1.7.9.5

